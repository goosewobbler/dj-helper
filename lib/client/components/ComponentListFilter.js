"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var renderSearchButton = function () { return (React.createElement("button", { className: "search-btn", key: "search-button" },
    React.createElement("svg", { xmlns: "http://www.w3.org/2000/svg", width: "16", height: "16", viewBox: "0 0 32 32" },
        React.createElement("path", { d: "M32 28.5l-8.2-8.2c3.4-5.1 2.9-12-1.6-16.4C19.7 1.3 16.3 0 13 0 9.7 0 6.3 1.3 3.8 3.8c-5.1 5.1-5.1 13.3 0 18.4C6.3 24.7 9.7 26 13 26c2.5 0 5.1-.7 7.3-2.2l8.2 8.2 3.5-3.5zM6.6 19.4C4.9 17.7 4 15.4 4 13s.9-4.7 2.6-6.4C8.3 4.9 10.6 4 13 4c2.4 0 4.7.9 6.4 2.6 3.5 3.5 3.5 9.2 0 12.7-1.7 1.7-4 2.6-6.4 2.6s-4.7-.8-6.4-2.5z" })))); };
var renderClearButton = function (onClick) { return (React.createElement("button", { className: "clear-btn", key: "clear-button", onClick: onClick },
    React.createElement("svg", { xmlns: "http://www.w3.org/2000/svg", width: "16", height: "16", viewBox: "0 0 32 32" },
        React.createElement("path", { d: "M32 3.5L28.5 0 16 12.5 3.5 0 0 3.5 12.5 16 0 28.5 3.5 32 16 19.5 28.5 32l3.5-3.5L19.5 16z" })))); };
var ComponentListFilter = /** @class */ (function (_super) {
    __extends(ComponentListFilter, _super);
    function ComponentListFilter(props) {
        var _this = _super.call(this, props) || this;
        _this.state = {
            filter: '',
            focussed: false,
        };
        return _this;
    }
    ComponentListFilter.prototype.render = function () {
        var _this = this;
        return (React.createElement("div", null,
            React.createElement("input", { value: this.state.filter, id: "search-input", placeholder: "Search", onKeyDown: function (event) { return _this.onKeyDown(event); }, onFocus: function () { return _this.onFocussed(true); }, onBlur: function () { return _this.onFocussed(false); }, onChange: function (event) { return _this.onInput(event); } }),
            this.renderIcon()));
    };
    ComponentListFilter.prototype.onFocussed = function (focussed) {
        this.setState({
            focussed: focussed,
        });
    };
    ComponentListFilter.prototype.onInput = function (event) {
        this.setState({
            filter: event.target.value,
        });
        this.props.onInput(event.target.value);
    };
    ComponentListFilter.prototype.onKeyDown = function (event) {
        if (event.keyCode === 27) {
            this.clearSearchInput();
        }
    };
    ComponentListFilter.prototype.clearSearchInput = function () {
        this.setState({
            filter: '',
        });
        this.props.onInput('');
    };
    ComponentListFilter.prototype.renderIcon = function () {
        var _this = this;
        return this.state.filter.length > 0 ? renderClearButton(function () { return _this.clearSearchInput(); }) : renderSearchButton();
    };
    return ComponentListFilter;
}(React.Component));
exports.default = ComponentListFilter;
//# sourceMappingURL=ComponentListFilter.js.map