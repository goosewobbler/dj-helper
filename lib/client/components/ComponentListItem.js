"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var ComponentState_1 = require("../../types/ComponentState");
var ListItemButton_1 = require("../styled/ListItemButton");
var ExternalLink_1 = require("./ExternalLink");
var IconButton_1 = require("./IconButton");
var createID = function (componentName) { return "component-list-item-" + componentName; };
var backgroundColorForState = function (state) {
    switch (state) {
        case ComponentState_1.default.Linking:
        case ComponentState_1.default.Starting:
            return 'orange';
        case ComponentState_1.default.Installing:
            return 'rgb(255, 90, 90)';
        case ComponentState_1.default.Building:
            return '#cb75ff';
        case ComponentState_1.default.Running:
            return '#baecff';
        default:
            break;
    }
    return 'white';
};
var renderFavouriteButton = function (props) {
    if (props.component.favorite) {
        return (React.createElement(IconButton_1.default, { className: "unfavorite-button", label: "Unfavorite", image: "/image/icon/favourited.svg", onClick: function () { return props.onToggleFavourite(); } }));
    }
    return (React.createElement(IconButton_1.default, { className: "favorite-button", label: "Favorite", image: "/image/icon/unfavourited.svg", onClick: function () { return props.onToggleFavourite(); } }));
};
var renderLaunchButton = function (component) {
    var link = "" + component.url + (Array.isArray(component.history) && component.history.length > 0
        ? component.history[0]
        : '');
    if (component.state === ComponentState_1.default.Running) {
        return (React.createElement(ExternalLink_1.default, { className: "launch-button", label: "Launch", link: link, color: "#2491c8", height: "21", padding: "0 4px" }));
    }
    return null;
};
var renderStartStopButton = function (props) {
    switch (props.component.state) {
        case ComponentState_1.default.Stopped:
            return (React.createElement(IconButton_1.default, { className: "start-button", label: "Start", image: "/image/icon/gel-icon-play.svg", onClick: props.onStart }));
        case ComponentState_1.default.Installing:
            return React.createElement(IconButton_1.default, { label: "Installing", loading: true });
        case ComponentState_1.default.Building:
            return React.createElement(IconButton_1.default, { label: "Building", loading: true });
        case ComponentState_1.default.Linking:
            return React.createElement(IconButton_1.default, { label: "Linking", loading: true });
        case ComponentState_1.default.Starting:
            return React.createElement(IconButton_1.default, { label: "Starting", loading: true });
        case ComponentState_1.default.Running:
            return (React.createElement(IconButton_1.default, { className: "stop-button", label: "Stop", image: "/image/icon/gel-icon-pause.svg", onClick: props.onStop }));
    }
};
var ComponentListItem = /** @class */ (function (_super) {
    __extends(ComponentListItem, _super);
    function ComponentListItem() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    ComponentListItem.prototype.shouldComponentUpdate = function (nextProps) {
        if (this.props.component !== nextProps.component) {
            return true;
        }
        if (this.props.selected !== nextProps.selected) {
            return true;
        }
        return false;
    };
    ComponentListItem.prototype.render = function () {
        var name = this.props.component.highlighted && this.props.component.highlighted.length > 0
            ? this.props.component.highlighted
            : this.props.component.displayName;
        return (React.createElement(ListItemButton_1.default, { role: "button", id: createID(this.props.component.name), backgroundColor: backgroundColorForState(this.props.component.state), highlighted: this.props.selected, onClick: this.props.onClick },
            renderFavouriteButton(this.props),
            React.createElement("span", { className: "component-name-label" }, name),
            renderLaunchButton(this.props.component),
            renderStartStopButton(this.props)));
    };
    return ComponentListItem;
}(React.PureComponent));
exports.default = ComponentListItem;
//# sourceMappingURL=ComponentListItem.js.map