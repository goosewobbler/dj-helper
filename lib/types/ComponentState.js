"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var ComponentState;
(function (ComponentState) {
    ComponentState[ComponentState["Stopped"] = 1] = "Stopped";
    ComponentState[ComponentState["Starting"] = 2] = "Starting";
    ComponentState[ComponentState["Installing"] = 3] = "Installing";
    ComponentState[ComponentState["Building"] = 4] = "Building";
    ComponentState[ComponentState["Running"] = 5] = "Running";
    ComponentState[ComponentState["Linking"] = 6] = "Linking";
})(ComponentState || (ComponentState = {}));
exports.default = ComponentState;
//# sourceMappingURL=ComponentState.js.map