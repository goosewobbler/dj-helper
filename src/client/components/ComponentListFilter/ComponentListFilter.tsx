import * as React from 'react';

const renderSearchButton = () => (
  <button className="search-btn" key="search-button">
    <svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" viewBox="0 0 32 32">
      <path d="M32 28.5l-8.2-8.2c3.4-5.1 2.9-12-1.6-16.4C19.7 1.3 16.3 0 13 0 9.7 0 6.3 1.3 3.8 3.8c-5.1 5.1-5.1 13.3 0 18.4C6.3 24.7 9.7 26 13 26c2.5 0 5.1-.7 7.3-2.2l8.2 8.2 3.5-3.5zM6.6 19.4C4.9 17.7 4 15.4 4 13s.9-4.7 2.6-6.4C8.3 4.9 10.6 4 13 4c2.4 0 4.7.9 6.4 2.6 3.5 3.5 3.5 9.2 0 12.7-1.7 1.7-4 2.6-6.4 2.6s-4.7-.8-6.4-2.5z" />
    </svg>
  </button>
);

const renderClearButton = (onClick: () => void) => (
  <button className="clear-btn" key="clear-button" onClick={onClick}>
    <svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" viewBox="0 0 32 32">
      <path d="M32 3.5L28.5 0 16 12.5 3.5 0 0 3.5 12.5 16 0 28.5 3.5 32 16 19.5 28.5 32l3.5-3.5L19.5 16z" />
    </svg>
  </button>
);

interface ComponentListFilterProps {
  onInput(filter: string): null;
}

interface ComponentListFilterState {
  filter: string;
  focussed: boolean;
}

class ComponentListFilter extends React.Component<ComponentListFilterProps, ComponentListFilterState> {
  constructor(props: ComponentListFilterProps) {
    super(props);
    this.state = {
      filter: '',
      focussed: false,
    };
  }

  public render() {
    return (
      <div>
        <input
          value={this.state.filter}
          id="search-input"
          placeholder="Search"
          onKeyDown={event => this.onKeyDown(event)}
          onFocus={() => this.onFocussed(true)}
          onBlur={() => this.onFocussed(false)}
          onChange={event => this.onInput(event)}
        />
        {this.renderIcon()}
      </div>
    );
  }

  private onFocussed(focussed: boolean) {
    this.setState({
      focussed,
    });
  }

  private onInput(event: any) {
    this.setState({
      filter: event.target.value,
    });
    this.props.onInput(event.target.value);
  }

  private onKeyDown(event: React.KeyboardEvent<HTMLInputElement>) {
    if (event.keyCode === 27) {
      this.clearSearchInput();
    }
  }

  private clearSearchInput() {
    this.setState({
      filter: '',
    });
    this.props.onInput('');
  }

  private renderIcon() {
    return this.state.filter.length > 0 ? renderClearButton(() => this.clearSearchInput()) : renderSearchButton();
  }
}

export default ComponentListFilter;
