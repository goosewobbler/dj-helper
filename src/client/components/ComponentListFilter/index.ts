import { connect } from 'react-redux';
import ComponentListFilter from './ComponentListFilter';

import { filterComponents } from '../../actions/components';

const mapStateToProps = () => ({});

const mapDispatchToProps = (dispatch: any) => ({
  onInput: (filter: string) => dispatch(filterComponents(filter)),
});

const Container = connect(
  mapStateToProps,
  mapDispatchToProps,
)(ComponentListFilter);

export default Container;
