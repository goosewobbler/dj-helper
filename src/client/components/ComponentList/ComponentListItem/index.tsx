import * as React from 'react';
import ExternalLink from '../../ExternalLink';
import LoadingIcon from '../../LoadingIcon';
import PauseIcon from './PauseIcon';
import PlayIcon from './PlayIcon';
import StarIcon from './StarIcon';
import IconButton from './IconButton';
import { ComponentData, ComponentState } from '../../../../common/types';

const createID = (componentName: string): string => `component-list-item-${componentName}`;

interface ComponentListItemProps {
  component: ComponentData;
  selected: boolean;
  onFavourite?(name: string, favourite: boolean): () => void;
  onClick?(name: string): void;
  onToggleFavourite?(): void;
  onStart?(name: string): () => void;
  onStop?(name: string): () => void;
}

const renderFavouriteButton = ({
  component: { favourite, name },
  onFavourite,
}: ComponentListItemProps): React.ReactElement => {
  if (favourite) {
    return (
      <IconButton className="unfavourite-button" label="Unfavourite" onClick={(): Function => onFavourite(name, false)}>
        <StarIcon starred />
      </IconButton>
    );
  }
  return (
    <IconButton className="favourite-button" label="Favourite" onClick={(): Function => onFavourite(name, true)}>
      <StarIcon starred={false} />
    </IconButton>
  );
};

const renderLaunchButton = (component: ComponentData): React.ReactElement => {
  const link = `${component.url}${
    Array.isArray(component.history) && component.history.length > 0 ? component.history[0] : ''
  }`;

  if (component.state === ComponentState.Running) {
    return (
      <ExternalLink className="launch-button" label="Launch" link={link} color="#2491c8" height="21" padding="0 4px" />
    );
  }
  return null;
};

const renderStartStopButton = (
  { component: { state } }: ComponentListItemProps,
  handleStart: () => void,
  handleStop: () => void,
): React.ReactElement => {
  switch (state) {
    case ComponentState.Stopped:
      return (
        <IconButton className="start-button" label="Start" onClick={handleStart}>
          <PlayIcon />
        </IconButton>
      );
    case ComponentState.Installing:
      return (
        <IconButton label="Installing">
          <LoadingIcon />
        </IconButton>
      );
    case ComponentState.Building:
      return (
        <IconButton label="Building">
          <LoadingIcon />
        </IconButton>
      );
    case ComponentState.Linking:
      return (
        <IconButton label="Linking">
          <LoadingIcon />
        </IconButton>
      );
    case ComponentState.Starting:
      return (
        <IconButton label="Starting">
          <LoadingIcon />
        </IconButton>
      );
    case ComponentState.Running:
      return (
        <IconButton className="stop-button" label="Stop" onClick={handleStop}>
          <PauseIcon />
        </IconButton>
      );
    default:
      return null;
  }
};

class ComponentListItem extends React.PureComponent<ComponentListItemProps> {
  public static displayName = 'ComponentListItem';

  private handleClick: () => void;

  private handleStart: () => void;

  private handleStop: () => void;

  public constructor(props: ComponentListItemProps) {
    super(props);
    const {
      component: { name },
      onClick,
      onStart,
      onStop,
    } = this.props;
    this.handleClick = (): void => onClick(name);
    this.handleStart = (): Function => onStart(name);
    this.handleStop = (): Function => onStop(name);
  }

  public shouldComponentUpdate(nextProps: ComponentListItemProps): boolean {
    const { component, selected } = this.props;
    const { highlighted, displayName, name, url, favourite, state } = component;
    if (
      // TODO: sort out this piece of shit
      name !== nextProps.component.name ||
      displayName !== nextProps.component.displayName ||
      highlighted !== nextProps.component.highlighted ||
      url !== nextProps.component.url ||
      favourite !== nextProps.component.favourite ||
      state !== nextProps.component.state ||
      selected !== nextProps.selected
    ) {
      return true;
    }
    return false;
  }

  public render(): React.ReactElement {
    const { selected, component } = this.props;
    const { highlighted, displayName, name } = component;
    const labelText: React.ReactElement[] | string = highlighted && highlighted.length > 0 ? highlighted : displayName;

    return (
      <div role="button" id={createID(name)} onClick={this.handleClick}>
        {renderFavouriteButton(this.props)}
        <span className="component-name-label">{labelText}</span>
        {renderLaunchButton(component)}
        {renderStartStopButton(this.props, this.handleStart, this.handleStop)}
      </div>
    );
  }
}

export { ComponentListItem, ComponentListItemProps };
