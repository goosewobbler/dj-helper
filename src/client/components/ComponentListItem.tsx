import * as React from 'react';

import ComponentState from '../../types/ComponentState';
import ComponentData from '../../types/ComponentData';
import ExternalLink from '../ui/ExternalLink';
import LoadingIcon from '../ui/icon/LoadingIcon';
import PauseIcon from '../ui/icon/PauseIcon';
import PlayIcon from '../ui/icon/PlayIcon';
import StarIcon from '../ui/icon/StarIcon';
import IconButton from './IconButton';

const createID = (componentName: string) => `component-list-item-${componentName}`;

interface IComponentListItemProps {
  component: ComponentData;
  selected: boolean;
  onClick?(): null;
  onToggleFavourite?(): null;
  onStart?(): null;
  onStop?(): null;
}

const renderFavouriteButton = (props: IComponentListItemProps) => {
  if (props.component.favorite) {
    return (
      <IconButton className="unfavorite-button" label="Unfavorite" onClick={() => props.onFavourite(props.name, false)}>
        <StarIcon starred />
      </IconButton>
    );
  }
  return (
    <IconButton className="favorite-button" label="Favorite" onClick={() => props.onFavourite(props.name, true)}>
      <StarIcon starred={false} />
    </IconButton>
  );
};

const renderLaunchButton = (component: ComponentData) => {
  const link = `${component.url}${
    Array.isArray(component.history) && component.history.length > 0 ? component.history[0] : ''
  }`;

  if (component.state === ComponentState.Running) {
    return (
      <ExternalLink className="launch-button" label="Launch" link={link} color="#2491c8" height="21" padding="0 4px" />
    );
  }
  return null;
};

const renderStartStopButton = (props: IComponentListItemProps, handleStart: () => void, handleStop: () => void) => {
  switch (props.state) {
    case ComponentState.Stopped:
      return (
        <IconButton className="start-button" label="Start" onClick={handleStart}>
          <PlayIcon />
        </IconButton>
      );
    case ComponentState.Installing:
      return (
        <IconButton label="Installing">
          <LoadingIcon />
        </IconButton>
      );
    case ComponentState.Building:
      return (
        <IconButton label="Building">
          <LoadingIcon />
        </IconButton>
      );
    case ComponentState.Linking:
      return (
        <IconButton label="Linking">
          <LoadingIcon />
        </IconButton>
      );
    case ComponentState.Starting:
      return (
        <IconButton label="Starting">
          <LoadingIcon />
        </IconButton>
      );
    case ComponentState.Running:
      return (
        <IconButton className="stop-button" label="Stop" onClick={handleStop}>
          <PauseIcon />
        </IconButton>
      );
  }
};

class ComponentListItem extends React.PureComponent<IComponentListItemProps> {
  public static displayName = 'ComponentListItem';

  private handleClick: () => void;
  private handleStart: () => void;
  private handleStop: () => void;

  constructor(props: IComponentListItemProps) {
    super(props);
    this.handleClick = () => this.props.onClick(this.props.name);
    this.handleStart = () => this.props.onStart(this.props.name);
    this.handleStop = () => this.props.onStop(this.props.name);
  }

  public shouldComponentUpdate(nextProps: IComponentListItemProps, nextState: {}, context: any) {
    if (
      this.props.name !== nextProps.name ||
      this.props.displayName !== nextProps.displayName ||
      this.props.highlighted !== nextProps.highlighted ||
      this.props.url !== nextProps.url ||
      this.props.favourite !== nextProps.favourite ||
      this.props.state !== nextProps.state ||
      this.props.selected !== nextProps.selected
    ) {
      return true;
    }
    return false;
  }

  public render() {
    const name: any =
      this.props.highlighted && this.props.highlighted.length > 0 ? this.props.highlighted : this.props.displayName;

    return (
      <div
        theme={this.props.theme}
        role="button"
        id={createID(this.props.name)}
        highlighted={this.props.selected}
        onClick={this.handleClick}
      >
        {renderFavouriteButton(this.props)}
        <span className="component-name-label">{name}</span>
        {renderLaunchButton(this.props.component)}
        {renderStartStopButton(this.props, this.handleStart, this.handleStop)}
      </div>
    );
  }
}

export default ComponentListItem;
