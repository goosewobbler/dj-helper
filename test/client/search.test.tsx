import toJson from 'enzyme-to-json';
import ComponentListFilter from '../../src/client/components/ComponentListFilter';
import ComponentListItem from '../../src/client/components/ComponentListItem';
import ComponentState from '../../src/types/ComponentState';
import IComponentData from '../../src/types/IComponentData';
import render from './mocks/app';

test('the filter has the correct style with no input.', () => {
  const { wrapper } = render();
  const filter = wrapper.find(ComponentListFilter);

  expect(toJson(filter)).toMatchSnapshot();
});

describe('when I focus and type in the filter.', () => {
  let wrapper: any;

  beforeEach(() => {
    wrapper = render().wrapper;

    const input = wrapper.find(ComponentListFilter).find('input');
    input.simulate('focus');
    input.simulate('keyDown');
    input.simulate('change', { target: { value: 'sport' } });
  });

  test('the filter has the correct style with input.', () => {
    const filter = wrapper.find(ComponentListFilter);

    expect(toJson(filter)).toMatchSnapshot();
  });

  test('the filter has the correct style when not focussed.', () => {
    const filter = wrapper.find(ComponentListFilter);
    const input = wrapper.find(ComponentListFilter).find('input');
    input.simulate('blur');

    expect(toJson(filter)).toMatchSnapshot();
  });

  test('only the matching results are displayed.', () => {
    const results = wrapper.find('.component-name-label').map((item: any) => item.text());

    expect(results).toEqual(['sport-search']);
  });

  test('pressing the clear button removes the filter.', () => {
    const clearButton = wrapper.find(ComponentListFilter).find('button');

    clearButton.simulate('click');

    const results = wrapper
      .find(ComponentListItem)
      .find('.component-name-label')
      .map((item: any) => item.text());

    expect(results).toEqual(['football-scores-view', 'ideas-page', 'sport-search', 'xyz-node-10']);
  });

  test('pressing the escape key removes the filter.', () => {
    const input = wrapper.find(ComponentListFilter).find('input');

    input.simulate('keyDown', { keyCode: 27 });

    const results = wrapper
      .find(ComponentListItem)
      .find('.component-name-label')
      .map((item: any) => item.text());

    expect(results).toEqual(['football-scores-view', 'ideas-page', 'sport-search', 'xyz-node-10']);
  });
});

test('the number of search results is capped.', () => {
  const components: IComponentData[] = [];
  for (let i = 0; i < 60; i++) {
    components[i] = {
      displayName: 'abc',
      favorite: false,
      name: 'bbc-morph-abc',
      rendererType: '0.12',
      state: ComponentState.Stopped,
      useCache: false,
      versions: {
        int: '',
        live: '',
        local: '',
        test: '',
      },
    };
  }

  const { wrapper } = render(components);
  const input = wrapper.find(ComponentListFilter).find('input');
  input.simulate('change', { target: { value: 'ab' } });
});
