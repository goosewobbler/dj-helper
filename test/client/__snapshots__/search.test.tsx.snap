// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the filter has the correct style with no input. 1`] = `
.glamor-2,
[data-glamor-2] {
  align-items: center;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-shrink: 0;
  margin: 8px;
  padding: 4px;
  transition: border-bottom 0.2s;
  border-bottom: solid 2px rgba(0, 0, 0, 0.19999999999999996);
  -webkit-box-align: center;
  -webkit-align-items: center;
  -webkit-flex-shrink: 0;
  -webkit-transition: border-bottom 0.2s;
  -moz-transition: border-bottom 0.2s;
}

.glamor-0,
[data-glamor-0] {
  background: transparent;
  border: none;
  flex-grow: 1;
  font-size: 20px;
  height: 24px;
  outline: none;
  -webkit-flex-grow: 1;
}

.glamor-1,
[data-glamor-1] {
  background: none;
  border: none;
  height: 16px;
  width: 16px;
  cursor: default;
}

<ComponentListFilter
  onInput={[Function]}
  theme={Object {}}
>
  <glamorous(div)
    focussed={false}
    theme={Object {}}
  >
    <div
      className="glamor-2"
    >
      <glamorous(input)
        id="search-input"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        placeholder="Search"
        theme={Object {}}
        value=""
      >
        <input
          className="glamor-0"
          id="search-input"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder="Search"
          value=""
        />
      </glamorous(input)>
      <glamorous(button)
        pointer={false}
      >
        <button
          className="glamor-1"
        >
          <svg
            height="16"
            viewBox="0 0 32 32"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M32 28.5l-8.2-8.2c3.4-5.1 2.9-12-1.6-16.4C19.7 1.3 16.3 0 13 0 9.7 0 6.3 1.3 3.8 3.8c-5.1 5.1-5.1 13.3 0 18.4C6.3 24.7 9.7 26 13 26c2.5 0 5.1-.7 7.3-2.2l8.2 8.2 3.5-3.5zM6.6 19.4C4.9 17.7 4 15.4 4 13s.9-4.7 2.6-6.4C8.3 4.9 10.6 4 13 4c2.4 0 4.7.9 6.4 2.6 3.5 3.5 3.5 9.2 0 12.7-1.7 1.7-4 2.6-6.4 2.6s-4.7-.8-6.4-2.5z"
            />
          </svg>
        </button>
      </glamorous(button)>
    </div>
  </glamorous(div)>
</ComponentListFilter>
`;

exports[`when I focus and type in the filter. the filter has the correct style when not focussed. 1`] = `
.glamor-2,
[data-glamor-2] {
  align-items: center;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-shrink: 0;
  margin: 8px;
  padding: 4px;
  transition: border-bottom 0.2s;
  border-bottom: solid 2px rgba(0, 0, 0, 0.19999999999999996);
  -webkit-box-align: center;
  -webkit-align-items: center;
  -webkit-flex-shrink: 0;
  -webkit-transition: border-bottom 0.2s;
  -moz-transition: border-bottom 0.2s;
}

.glamor-0,
[data-glamor-0] {
  background: transparent;
  border: none;
  flex-grow: 1;
  font-size: 20px;
  height: 24px;
  outline: none;
  -webkit-flex-grow: 1;
}

.glamor-1,
[data-glamor-1] {
  background: none;
  border: none;
  height: 16px;
  width: 16px;
  cursor: pointer;
}

<ComponentListFilter
  onInput={[Function]}
  theme={Object {}}
>
  <glamorous(div)
    focussed={false}
    theme={Object {}}
  >
    <div
      className="glamor-2"
    >
      <glamorous(input)
        id="search-input"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        placeholder="Search"
        theme={Object {}}
        value="sport"
      >
        <input
          className="glamor-0"
          id="search-input"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder="Search"
          value="sport"
        />
      </glamorous(input)>
      <glamorous(button)
        onClick={[Function]}
        pointer={true}
      >
        <button
          className="glamor-1"
          onClick={[Function]}
        >
          <svg
            height="16"
            viewBox="0 0 32 32"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M32 3.5L28.5 0 16 12.5 3.5 0 0 3.5 12.5 16 0 28.5 3.5 32 16 19.5 28.5 32l3.5-3.5L19.5 16z"
            />
          </svg>
        </button>
      </glamorous(button)>
    </div>
  </glamorous(div)>
</ComponentListFilter>
`;

exports[`when I focus and type in the filter. the filter has the correct style with input. 1`] = `
.glamor-0,
[data-glamor-0] {
  background: transparent;
  border: none;
  flex-grow: 1;
  font-size: 20px;
  height: 24px;
  outline: none;
  -webkit-flex-grow: 1;
}

.glamor-2,
[data-glamor-2] {
  align-items: center;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-shrink: 0;
  margin: 8px;
  padding: 4px;
  transition: border-bottom 0.2s;
  border-bottom: solid 2px undefined;
  -webkit-box-align: center;
  -webkit-align-items: center;
  -webkit-flex-shrink: 0;
  -webkit-transition: border-bottom 0.2s;
  -moz-transition: border-bottom 0.2s;
}

.glamor-1,
[data-glamor-1] {
  background: none;
  border: none;
  height: 16px;
  width: 16px;
  cursor: pointer;
}

<ComponentListFilter
  onInput={[Function]}
  theme={Object {}}
>
  <glamorous(div)
    focussed={true}
    theme={Object {}}
  >
    <div
      className="glamor-2"
    >
      <glamorous(input)
        id="search-input"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        placeholder="Search"
        theme={Object {}}
        value="sport"
      >
        <input
          className="glamor-0"
          id="search-input"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder="Search"
          value="sport"
        />
      </glamorous(input)>
      <glamorous(button)
        onClick={[Function]}
        pointer={true}
      >
        <button
          className="glamor-1"
          onClick={[Function]}
        >
          <svg
            height="16"
            viewBox="0 0 32 32"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M32 3.5L28.5 0 16 12.5 3.5 0 0 3.5 12.5 16 0 28.5 3.5 32 16 19.5 28.5 32l3.5-3.5L19.5 16z"
            />
          </svg>
        </button>
      </glamorous(button)>
    </div>
  </glamorous(div)>
</ComponentListFilter>
`;
